theano/tensor/nnet/abstract_conv.py

```diff
@@ -1517,18 +1525,23 @@ def conv(self, img, kern, mode="valid", dilation=1):
                      ] = kern
 
         if self.convdim == 2:
+            if img.shape[1] % self.num_groups != 0:
+                raise ValueError("Number of input channels must be divisible by num_groups")
             val = _valfrommode(mode)
             bval = _bvalfromboundary('fill')
+            input_channel_offset = img.shape[1] // self.num_groups
+            output_channel_offset = kern.shape[0]
```

Comment:
Adding to this, I think there were two ways for this to be implemented.

The first method takes input of shape `(batches, in_channels, ih, iw )` and kernels of shape `(num_filters, in_channels, kh,  kw)` and `num_groups` and splits `in_channels` into `num_groups` different groups, giving inputs of each group shape 
`(batches, in_channels // num_groups , ih, iw)` and kernel of shape `(num_filters, in_channels // num_groups , kh, kw)`. The final output of shape will be` (batches, num_filters * num_groups , oh, ow)` .
This method would require the user be aware of the fact that total number of output channels is `num_filters * num_groups`.  This is similar to the tensorflow depthwise_convolution implementation (there num_groups = in_channels).

The other implementation would be to take inputs of shape `(batches, in_channels, ih, iw)` and kernel of shape `(num_filters, in_channels // num_groups , kh, kw)` then after splitting, the inputs would be of shape `(batches, in_channels // num_groups, ih, iw)` and kernel would be of shape `(num_filters // num_groups , in_channels // num_groups , kh , kw)` the final output would be of shape `(batches , out_channels, oh , ow)` .This method would require the user to compute the shape of the kernel taking `num_groups` into account i.e the 2nd dimension of the kernel to be exactly `in_channels // num_groups` .Also `num_filters` must be divisible by `num_groups`.

Currently, the first method is implemented. But it should not take me more than a day to shift it to implement the second method instead if that is preferred